### PR TITLE
Feature/redux done

### PR DESCRIPTION
**Pull request by eugene-nosenko**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -30,5 +27,6 @@
       To begin the development, run `npm start` in this folder.
       To create a production bundle, use `npm run build`.
     -->
-  </body>
-</html>
+</body>
+
+</html>
```
Comment (MariannaMilovanova) [inline public/index.html]: maybe eslint this file

File: src/EmojiResultRow.js [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
 import PropTypes from "prop-types";
+import Favorite from "./Favorite";
 import "./EmojiResultRow.css";
 
 const EmojiResultsRow = props => {
-  const { symbol, title } = props;
-  const codePointHex = symbol.codePointAt(0).toString(16);
+  const { symbol, title, addToFavorite, removeFromFavorite, codePointHex, isFavorite } = props;
+
   const src = `//cdn.jsdelivr.net/emojione/assets/png/${codePointHex}.png`;
```
Comment (MariannaMilovanova) [inline src/EmojiResultRow.js:8]: you can also pass src from EmodjiResult file

File: src/EmojiResultRow.js [hunk 3]
```diff
@@ -1,15 +1,24 @@
...
     <div className="component-emoji-result-row copy-to-clipboard" data-clipboard-text={symbol}>
+      <Favorite
+        isFavorite={isFavorite}
+        codePointHex={codePointHex}
+        addToFavorite={addToFavorite}
+        removeFromFavorite={removeFromFavorite}
+      />
       <img alt={title} src={src} />
+      <span className="title">{symbol.codePointAt(0)} </span>
+      <span className="title">{codePointHex} </span>
```
Comment (MariannaMilovanova) [inline src/EmojiResultRow.js:19]: are you sure that this html layout was required?

File: src/EmojiResults.js [hunk 4]
```diff
@@ -5,7 +5,13 @@ import Clipboard from "clipboard";
...
 import "./EmojiResults.css";
 
-const EmojiResults = ({ emojiData }) => {
+import { connect } from "react-redux";
+import { addToFavorite, removeFromFavorite } from "./store/actions/favorite";
+
+const EmojiResults = props => {
+  const { emojiData, favorite, addToFavorite, removeFromFavorite } = props;
+  console.log(favorite);
+
```
Comment (MariannaMilovanova) [inline src/EmojiResults.js:12]: please remove console.log

File: src/Favorite.css [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+.favstar-true:before {
+  content: "⭐";
+  font-size: 2em;
+  color: transparent;
+  text-shadow: 0 0 0 Gold;
+  filter: Gold(100%);
+  padding-right: 10px;
+}
```
Comment (MariannaMilovanova) [inline src/Favorite.css]: please do not forget to add empty rows in the end of files, it's needed for better compilation

File: src/Favorite.js [hunk 6]
```diff
@@ -1 +1,26 @@
...
+
+    if (event.target.className === "favstar-true") {
+      removeFromFavorite(codePointHex);
+    }
+  };
+
+  return (
+    <>
+      <span onClick={handleFavorite}>
+        {isFavorite ? <i className="favstar-true" /> : <i className="favstar-false" />}
```
Comment (MariannaMilovanova) [inline src/Favorite.js]: you can make 2 onClick function
```
 <><span onClick={()=> addToFavorite(codePointHex)}><i /></span> 
 <span onClick={()=> removeFromFavorite(codePointHex)}><i/></span></>
```

File: src/emojiList.json [hunk 7]
```diff
@@ -9097,5 +9097,31 @@
...
+    "symbol": "🤽🏐",
+    "keywords": "Meet Parents movie"
+  },
+  {
+    "title": "Beauty and the Beast movie",
+    "symbol": "👸🏽🥀👹",
+    "keywords": "Beauty Beast movie"
   }
+
 ]
```
Comment (MariannaMilovanova) [inline src/emojiList.json]: empty row

**Review by MariannaMilovanova — APPROVED** [review]

File: src/Favorite.js (hunk 6 above)
Comment (eugene-finn) [reply src/Favorite.js:20]: это не работает
`import React from "react";
import "./Favorite.css";

const Favorite = props => {
  const { addToFavorite, removeFromFavorite, codePointHex, isFavorite } = props;

  return (
    <>
      {isFavorite ? (
        <span onClick={() => addToFavorite(codePointHex)}>
          <i className="favstar-true" />
        </span>
      ) : (
        <span onClick={() => removeFromFavorite(codePointHex)}>
          <i className="favstar-false" />
        </span>
      )}
    </>
  );
};

export default Favorite;
`

**Pull request merged**